packages/contracts/test/factory/dao-factory.ts

```diff
@@ -187,305 +211,142 @@ describe('DAOFactory: ', function () {
+        WITHDRAW_PERMISSION_ID,
         daoFactory.address,
-        createdDao.address,
-        voting.address,
+        dao,
+        dao,
         PermissionManagerAllowFlagAddress
-      )
-      .to.emit(createdDao, EVENTS.Granted)
+      );
+    await expect(tx)
+      .to.emit(daoContract, EVENTS.Granted)
       .withArgs(
-        MODIFY_ALLOWLIST_PERMISSION_ID,
+        UPGRADE_PERMISSION_ID,
         daoFactory.address,
-        createdDao.address,
-        voting.address,
+        dao,
+        dao,
         PermissionManagerAllowFlagAddress
-      )
-      .to.emit(createdDao, EVENTS.Revoked)
+      );
+    await expect(tx)
+      .to.emit(daoContract, EVENTS.Granted)
       .withArgs(
-        DAOPermissions[1],
+        SET_SIGNATURE_VALIDATOR_PERMISSION_ID,
         daoFactory.address,
+        dao,
+        dao,
+        PermissionManagerAllowFlagAddress
+      );
+    await expect(tx)
+      .to.emit(daoContract, EVENTS.Granted)
+      .withArgs(
+        SET_TRUSTED_FORWARDER_PERMISSION_ID,
         daoFactory.address,
-        createdDao.address
-      )
-      .to.emit(createdDao, EVENTS.Granted)
+        dao,
+        dao,
+        PermissionManagerAllowFlagAddress
+      );
+    await expect(tx)
+      .to.emit(daoContract, EVENTS.Granted)
       .withArgs(
-        EXECUTE_PERMISSION_ID,
+        SET_METADATA_PERMISSION_ID,
         daoFactory.address,
-        voting.address,
-        createdDao.address,
+        dao,
+        dao,
         PermissionManagerAllowFlagAddress
       );
-
-    // ===== Test if user can create a vote and execute it ======
-
-    // should be only callable by AllowlistVoting
-    await expect(createdDao.execute(0, [])).to.be.revertedWith(
-      customError(
-        'Unauthorized',
-        createdDao.address,
-        createdDao.address,
-        ownerAddress,
-        EXECUTE_PERMISSION_ID
-      )
-    );
-
-    await expect(voting.setConfiguration(1, 2, 3)).to.be.revertedWith(
-      customError(
-        'DaoUnauthorized',
-        createdDao.address,
-        voting.address,
-        voting.address,
-        ownerAddress,
-        SET_CONFIGURATION_PERMISSION_ID
-      )
-    );
-
-    const actions = [
-      {
-        to: actionExecuteContract.address,
-        value: 0,
-        data: actionExecuteContract.interface.encodeFunctionData('setTest', []),
-      },
-      {
-        to: voting.address,
-        value: 0,
-        data: voting.interface.encodeFunctionData(
-          'setConfiguration',
-          [3, 4, 5]
-        ),
-      },
-    ];
-
-    await voting.createVote('0x', actions, 0, 0, false, VoteOption.Yes);
-
-    expect(await voting.vote(0, VoteOption.Yes, true))
-      .to.emit(createdDao, EVENTS.Executed)
-      .withArgs(voting.address, 0, [], [])
-      .to.emit(voting, EVENTS.ConfigUpdated)
-      .withArgs(3, 4, 5);
-
-    expect(await actionExecuteContract.test()).to.equal(true);
   });
+
+  // TODO: test erc20Voting, and allowlistVoting when ready.
```

Comment:
Is this a task? I would remove the TODO and instead create empty test that fail but are skipped.

Reply:
once the governance setups are ready they will have their own tests, so it should be fine

Reply:
I would add skipped, empty, failing tests but this should not prevent us from merging. Go ahead if you wish